asmcomp/amd64/emit.mlp

```diff
@@ -544,10 +544,21 @@ let emit_instr fallthrough i =
   | Lop(Iconst_int n) ->
       if n = 0n then begin
         match i.res.(0).loc with
-        | Reg _ -> I.xor (res i 0) (res i 0)
-        | _     -> I.mov (int 0) (res i 0)
-      end
-      else
+        | Reg _ ->
+          (* Clearing the bottom half also clears the top half (except for
+             64-bit-only registers where the behaviour is as if the operands
+             were 64 bit). *)
+          I.xor (res32 i 0) (res32 i 0)
+        | _ ->
+          I.mov (int 0) (res i 0)
+      end else if (n > 0n) && (n <= 0xFFFF_FFFFn) then begin
```

Comment:
The house style is to not put parentheses around `n > 0n` and `n <= 0xFFFF_FFFFn`, because they are not needed and nothing is ambiguous.

Reply:
certainly, now fixed!